.github/actions/publish-nuget/action.yml

```
@@ -60,10 +60,16 @@ runs:
         ${{ inputs.publish-version }}
         '@.Trim()
 
-        if ($version -eq 'USE_GITHUB_REF_NAME') { $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v") }
+        if ($version -eq 'USE_GITHUB_REF_NAME') {
+            $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v")
+            if (!($version -match '^v?(\d+\.\d+\.\d+)(-alpha\.\d+\.[a-z]{2,}-\d+)?$')) {
```

You don't need to group the match around the first three digits:
```suggestion
            if (!($version -match '^v?\d+\.\d+\.\d+(-alpha\.\d+\.[a-z]{2,}-\d+)?$')) {
```